signer/local_cosigner.go

```diff
@@ -85,54 +77,71 @@ type HrsMetadata struct {
 
 // LocalCosigner responds to sign requests using their share key
 // The cosigner maintains a watermark to avoid double-signing
-//
-// LocalCosigner signing is thread saafe
+// TODO: Clarify what you mean with cosinger here.
+// LocalCosigner signing is thread safe
 type LocalCosigner struct {
-	pubKeyBytes []byte
-	key         CosignerKey
-	rsaKey      rsa.PrivateKey
-	total       uint8
-	threshold   uint8
+	/*
+		//key         CosignerKey		- moved to threshold
+		//rsaKey      rsa.PrivateKey	- moved to threshold
+		//threshold uint8 				- moved to threshold
 
-	// stores the last sign state for a share we have fully signed
-	// incremented whenever we are asked to sign a share
-	lastSignState *SignState
+		// stores the last sign state for a share we have fully signed
+		// incremented whenever we are asked to sign a share
+		//lastSignState *SignState 		- moved to threshold
 
-	// signing is thread safe
-	lastSignStateMutex sync.Mutex
+		// signing is thread safe 		- moved to threshold
+		//lastSignStateMutex sync.Mutex - moved to threshold
 
-	// Height, Round, Step -> metadata
-	hrsMeta map[HRSTKey]HrsMetadata
-	peers   map[int]CosignerPeer
+		// Height, Round, Step -> metadata 	- moved to threshold
+		//hrsMeta map[HRSTKey]HrsMetadata 	- moved to threshold
 
-	address string
+	*/
+	pubKeyBytes []byte
+	total       uint8
+	address     string
+	//peers       map[int]CosignerPeer
+	localsigner *LocalSoftSignThresholdEd25519Signature
 }
 
 func (cosigner *LocalCosigner) SaveLastSignedState(signState SignStateConsensus) error {
-	return cosigner.lastSignState.Save(signState, &cosigner.lastSignStateMutex)
+	return cosigner.localsigner.LastSignState.Save(signState, &cosigner.localsigner.LastSignStateMutex)
 }
 
 func NewLocalCosigner(cfg LocalCosignerConfig) *LocalCosigner {
+	localsigner := LocalSoftSignThresholdEd25519Signature{
```

Comment:
This should be passed as a parameter in the `cfg` rather than constructed here. We could add a method in the new `LocalSoftSignThresholdEd25519Signature` file to return this config such as `func NewLocalThresholdEd25519Signature(key CosignerKey, rsaKey PrivateKey, total, threshold int) LocalSoftSignThresholdEd25519Signature {` which could then be called from `cmd/horcrux/cmd/cosigner.go`

Reply:
Work in progress. Have atleast added the constructor functions in threshold_ed25519_localsoftsign_signer. But still "initialises" (constructor function are called) in local_cosigner.